chainer/link.py

```diff
@@ -172,27 +240,16 @@ def add_param(self, name, shape=None, dtype=numpy.float32,
                 ignored.
 
         """
-        d = self.__dict__
-        if name in d:
-            raise AttributeError(
-                'cannot register a new parameter %s: attribute exists'
-                % name)
```

Comment:
The removal of this code changes the behavior of `add_param`.  Consider the following code:

```
l = chainer.Link()
l.add_persistent('w', np.array([1, 2, 3]))
l.add_param('w')
```

Previous code raised `AttributeError` but it does not if we apply this PR.